pydash/arrays.py

```diff
@@ -1322,6 +1348,31 @@ def sorted_index(array, value, callback=None):
     return bisect_left(array, value)
 
 
+def sorted_index_of(array, value):
+    """Returns the index of the matched `value` from the sorted `array`, else
+    -1.
+
+    Args:
+        array (list): Array to inspect.
+        value (mixed): Value to search for.
+
+    Returns:
+        int: Returns the index of the first matched value, else -1.
+
+    Example:
+
+        >>> sorted_index_of([10, 5, 7, 3], 3)
+        0
+        >>> sorted_index_of([10, 10, 5, 7, 3], 10)
+        3
+        >>> sorted_index_of([10, 5, 7, 3], 11)
+        -1
+
+    .. versionadded:: TODO
+    """
+    return sorted(array).index(value) if value in array else -1
```

Comment:
This isn't quite what this function is supposed to do as the output in your examples doesn't match the corresponding function calls in Lodash's `.sortedIndexOf`:

```javascript
> var lo = require('lodash');
> lo.sortedIndexOf([10, 5, 7, 3], 3)
-1
> lo.sortedIndexOf([10, 10, 5, 7, 3], 10)
-1
> lo.sortedIndexOf([10, 5, 7, 3], 11)
-1
```

If you look at the Lodash source, then you can see that Python equivalent is basically:

```python
index = sorted_index(array, value)
if index < len(array) and array[index] == value:
    return index
else:
    return -1
```

* https://github.com/lodash/lodash/blob/master/sortedIndexOf.js
* https://github.com/lodash/lodash/blob/master/sortedIndex.js
* https://github.com/lodash/lodash/blob/master/.internal/baseSortedIndex.js